src/core/ActivityInfoTree.h

```diff
@@ -59,12 +61,17 @@ protected slots:
 private:
     // this is the full activity list, it never changes
     QList<ActivityInfo *> m_menuTreeFull;
+    QList<ActivityInfo *> m_menuTreeFulltemp;
     // represents the Menu view and can be filtered
     QList<ActivityInfo *> m_menuTree;
+    QList<ActivityInfo *> m_searched;
     ActivityInfo *m_rootMenu;
     ActivityInfo *m_currentActivity;
     static int menuTreeCount(QQmlListProperty<ActivityInfo> *property);
     static ActivityInfo *menuTreeAt(QQmlListProperty<ActivityInfo> *property, int index);
+    QObject *m_mainWindow;
+    int templength = 0;
+    QString searchedtext;
```

Comment:
ex: while searching for tux 
time taken by cpu in micro-seconds
for:
t = 78
u = 82 
x= 20 
I don't have the code I was using before ,it used to give more convincing results(right now "u" is taking more time ,this shouldn't be happening but it's most probably because of time taken by other threads which are working simultaneously).

My  humble opinion is to use  optimization since its doesn't come with any bad consequence.I understand it's just 100 activities and even if we don't use optimization  time taken by cpu will be in micro-seconds which is not much  .

To me it's better to loop for less number of time on every iteration rather than looping for same number of time .

Correct me if am wrong anywhere :)